pyro/infer/reparam/conjugate.py

```diff
@@ -0,0 +1,88 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import pyro
+import pyro.distributions as dist
+
+from .reparam import Reparam
+
+
+class ConjugateReparam(Reparam):
+    """
+    Reparameterize to a conjugate updated distribution.
+
+    This updates a prior distribution ``fn`` using the
+    :meth:`~pyro.distributions.Distribution.conjugate_update`
+    method.  The guide may be either a distribution object or a callable
+    inputting model ``*args,**kwargs`` and returning a distribution object. The
+    guide may be approximate or learned.
+
+    For example consider the model and naive variational guide::
+
+        total = torch.tensor(10.)
+        count = torch.tensor(2.)
+
+        def model():
+            prob = pyro.sample("prob", dist.Beta(0.5, 1.5))
+            pyro.sample("count", dist.Binomial(total, prob), obs=count)
+
+        guide = AutoDiagonalNormal(model)  # learns the posterior over prob
+
+    Instead of using this learned guide, we can hand-compute the conjugate
+    posterior distribution over "prob", and then use a simpler guide during
+    inference, in this case an empty guide::
+
+        reparam_model = poutine.reparam(model, {
+            "prob": ConjugateReparam(dist.Beta(1 + count, 1 + total - count))
+        })
+
+        def reparam_guide():
+            pass  # nothing remains to be modeled!
+
+    :param guide: A likelihood distribution or a callable returning a
+        guide distribution. Only a few distributions are supported, depending
+        on the prior distribution's
+        :meth:`~pyro.distributions.Distribution.conjugate_update`
+        implementation.
+    :type guide: ~pyro.distributions.Distribution or callable
+    """
+    def __init__(self, guide):
+        self.guide = guide
```

Comment:
We do not currently need to call `super().__init__()` because `Reparam` has no `__init__()` method; however I could add `super().__init__()` calls to all `Reparam` subclasses if you would like.

Reply:
Thanks for explaining, we can probably add it later if `Reparam` gets an `__init__`. I was just wondering where `self.args_kwargs` was coming from, but I see that  it is being set in `ReparamMessenger`.